scripts/update-projects-version.py

```diff
@@ -0,0 +1,375 @@
+            f"{YAML_TAB*2}- pelorus-operator.v{version}\n"
+        )
+
+    with open(cluster_service_version_file, mode="w", encoding="utf-8") as file_content:
+        file_content.writelines(cluster_service_version_file_content)
+
+
+def update_operator_folder(next_operator_version: str, destination: Path) -> None:
+    logging.info("Creating operator init files")
+    # TODO run update command instead of init every time
+    run_command(
+        f"operator-sdk init --plugins=helm --domain={OPERATOR_DOMAIN} --project-name={OPERATOR_NAME}",
+        directory=destination,
+    )
+    logging.info("Creating operator api")
+    exporters_charts_in_operator = destination / "helm-charts/pelorus/charts/exporters"
+    exporters_charts_in_operator.mkdir(parents=True)
+    shutil.copytree(
+        PELORUS_EXPORTERS_FOLDER, exporters_charts_in_operator, dirs_exist_ok=True
+    )
+    run_command(
+        f"operator-sdk create api --helm-chart={PELORUS_CHARTS_FOLDER}",
+        directory=destination,
+    )
+    for file in exporters_charts_in_operator.parent.glob("*.tgz"):
+        file.unlink()
+
+    replace_in_file(
+        file=PELORUS_OPERATOR_FOLDER / "Makefile",
+        pattern=r"VERSION \?= 0\.0\.1",
+        new=f"VERSION ?= {next_operator_version}",
+        number_of_changes=1,
+    )
+    operator_image = f"quay.io/{OPERATOR_ORGANIZATION_NAME}/{OPERATOR_NAME}"
+    replace_in_file(
+        file=PELORUS_OPERATOR_FOLDER / "Makefile",
+        pattern=r"IMAGE_TAG_BASE \?= pelorus\.dora-metrics\.io/pelorus-operator",
+        new=f"IMAGE_TAG_BASE ?= {operator_image}",
+        number_of_changes=1,
+    )
+    logging.info("Generating operator kustomize manifests files")
+    run_command(
+        "operator-sdk generate kustomize manifests -q --interactive=false",
+        directory=destination,
+    )
+
+    for file in PATCHES_FOLDER.rglob("*"):
+        if file.is_file():
+            logging.info(f"Applying patch {file}")
+            run_command(f"patch -d {destination} -p0 < {file}")
+
+    replace_in_file(
+        file=PELORUS_OPERATOR_FOLDER
+        / "config/manifests/bases/pelorus-operator.clusterserviceversion.yaml",
+        pattern="    containerImage: placeholder",
+        new=f"    containerImage: {operator_image}:{next_operator_version}",
+        number_of_changes=1,
+    )
+    logging.info("Executing operator make bundle")
+    run_command("make bundle", directory=destination)
+
+    pelorus_operator_image_tags = request.Request(
+        "https://quay.io/api/v1/repository/pelorus/pelorus-operator/tag/"
+    )
+    pelorus_operator_image_tags.add_header("Authorization", "Bearer XYZ")
+    with request.urlopen(pelorus_operator_image_tags) as response:
+        tags: List[Dict[str, str]] = list(json.load(response)["tags"])
+    tag_names = [
+        tag["name"]
+        for tag in tags
+        if tag.get("end_ts") is None and not tag["name"].upper().isupper()
+    ]
+    tag_names.sort(reverse=True)
```

Comment:
This needs improving the logic, like first converting to a semver object, and then running sort